refactoring getAttrsForRequest to allow relationship updates

zachfedor opened the pull request:
Previously, this method assumed that the request body contained an `attributes` property which is fine when you are actually updating properties of a model (e.g. changing the name of a user or the content of a post).

My changes allow updating model relationships without modifying the existing models (e.g. changing a post's author to another user).

samselikoff commented:
Looks great! Could we get a test for this, one that fails on without your changes?

zachfedor commented:
Good call. I copied the previous unit test for `_getAttrsForRequest`, removed the attributes, and simplified it. It passes with my change and fails without it.

Hope this helps!

samselikoff commented:
Yes, perfect! Could you squash commits? Then this is ready to go

zachfedor commented:
Commits are squashed. I believe my PR will also do away with the need for [this one](https://github.com/samselikoff/ember-cli-mirage/pull/572)

samselikoff commented:
Nice! And yep, I agree. Thank you!
